sale_commission/models/account_invoice.py

```diff
@@ -112,39 +96,44 @@ class AccountInvoiceLineAgent(models.Model):
     commission = fields.Many2one(
         comodel_name="sale.commission", required=True, ondelete="restrict")
     amount = fields.Float(
-        string="Amount settled", compute="_get_amount", store=True)
+        string="Amount settled", compute="_compute_amount", store=True)
     agent_line = fields.Many2many(
         comodel_name='sale.commission.settlement.line',
         relation='settlement_agent_line_rel', column1='agent_line_id',
         column2='settlement_id')
-    settled = fields.Boolean(compute="_get_settled", store=True)
+    settled = fields.Boolean(compute="_compute_settled", store=True)
 
-    @api.one
     @api.onchange('agent')
     def onchange_agent(self):
         self.commission = self.agent.commission
 
-    @api.one
-    @api.depends('commission.commission_type', 'invoice_line.price_subtotal')
-    def _get_amount(self):
-        self.amount = 0.0
-        if (not self.invoice_line.product_id.commission_free and
-                self.commission):
-            subtotal = self.invoice_line.price_subtotal
-            if self.commission.commission_type == 'fixed':
-                self.amount = subtotal * (self.commission.fix_qty / 100.0)
-            else:
-                self.amount = self.commission.calculate_section(subtotal)
-
-    @api.one
+    @api.depends('commission.commission_type', 'invoice_line.price_subtotal',
+                 'commission.amount_base_type')
+    def _compute_amount(self):
```

Comment:
In general, I would not change method names of stable modules, as there could be other modules using/extending them, that would suddenly stop working